tck/src/main/java/org/eclipse/microprofile/config/tck/EmptyValuesTest.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2016-2017 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package org.eclipse.microprofile.config.tck;
+
+import org.eclipse.microprofile.config.Config;
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.testng.Arquillian;
+import org.jboss.shrinkwrap.api.Archive;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.EmptyAsset;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.testng.annotations.Test;
+
+import javax.inject.Inject;
+
+import static org.testng.Assert.assertEquals;
+
+public class EmptyValuesTest extends Arquillian {
+
+    public static final String EMPTY_PROPERTY = "my.empty.property";
+
+    @Deployment
+    public static Archive deployment() {
+        return ShrinkWrap.create(WebArchive.class)
+                .addClasses(EmptyValuesTest.class, EmptyValuesBean.class)
+                .addAsWebInfResource(EmptyAsset.INSTANCE, "beans.xml");
+    }
+
+    @Inject
+    private EmptyValuesBean emptyValuesBean;
+
+    @Inject
+    private Config config;
+
+    @Test
+    public void testEmptyStringValues() {
+        assertEquals(emptyValuesBean.getStringValue(), "");
+    }
+
+    @Test
+    public void testEmptyStringProgrammaticLookup() {
+        System.setProperty(EMPTY_PROPERTY, "");
+        String stringValue = config.getValue(EMPTY_PROPERTY, String.class);
```

Comment:
I prefer you add my.empty.property to microprofile-config.properties as it would represent the real use case. I think a lot of times to have them set blank and then override via a system property to set sometime more meaningful.

As the tck has the set the presence of using system property and expect it to appear as config straightaway. I am not going to block your commit though I think it can be approved. If @struberg or @OndrejM are fine with it, I am ok with it.